### PR TITLE
SCT event handling - self-analysis of tests based on events

### PR DESCRIPTION
**Pull request by larisau**
Pushed to scylladb:events branch as well.

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -301,7 +303,10 @@ def __init__(self, name, ssh_login_info=None, base_logdir=None, node_prefix=None
         self.cs_start_time = None
         self.database_log = os.path.join(self.logdir, 'database.log')
         self._database_log_errors_index = []
-        self._database_error_patterns = ['std::bad_alloc', 'integrity check failed']
+        self._database_error_patterns = ['std::bad_alloc', 'std::runtime_error', 'integrity check failed',
+                                         'stacktrace', 'segmentation', 'exception']
```
Comment (roydahan) [inline sdcm/cluster.py]: Add backtrace.

Comment (roydahan) [reply sdcm/cluster.py]: Add "reactor stall" (check the exact phrase)

**Comment by larisau** [comment]
Ready, tested on longevity short test.
TODO: change threshold for count of errors - trigger for test termination

**Comment by roydahan** [comment]
@larisau, @bentsi, I would like to merge this branch this week.
Originally, I thought we will test with this branch but this is not happening.
Let's rebase, review again to make sure nothing has changed since and merge.

**Comment by larisau** [comment]
@roydahan  rebased

**Comment by roydahan** [comment]
@bentsi if we won't merge this, we 'll keep having conflicts forever.

**Comment by bentsi** [comment]
I will review it tomorrow in the morning

File: sdcm/event.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+import requests
+import json
+from prometheus import SCTMetrics
+from avocado.utils import process
+
+import prometheus
+
+logger = logging.getLogger(__name__)
+
+COLOR = {'white':   "\033[0;37;60m",
```
Comment (bentsi) [inline sdcm/event.py]: enum

**Comment by amoskong** [comment]
@bentsi @roydahan We need to arrange someone to rebase it, or it's no longer relevant and we can close it?

**Comment by bentsi** [comment]
@amoskong I am going to work on the new SCT events feature next week (hopefully), so let's close this.
(@roydahan we have Larisa's code in `larisau:events`)

**Pull request closed without merging**